packages/libs/src/sdk/api/chats/ChatsApi.ts

```diff
@@ -853,6 +902,9 @@ export class ChatsApi
               created_at: data.metadata.timestamp
             }
           })
+        } else if (data.rpc.method === 'chat.blast') {
+          const userId = data.metadata.userId
+          await this.upgradeBlasts(userId)
```

Comment:
Wait can't we use `this.listenerUserId` here?

Reply:
oh man... it was not passed in `listen()` call in the middleware. i could set that there and then we wouldn't need to append the receiver user id to the rpc metadata. i was wondering why it was still undefined!!

but that means the chats api is more stateful... thoughts? i could go either way.

Reply:
Ohhh right right - forgot the websockets were associated via signature recovery.

I think either way is fine! Though I wonder if we should remove `this.listenerUserId` if it's not being used

Reply:
yeah.... i kinda like the stateless idea. i'll give that a whirl and maybe remove listenUserId in a follow-up PR

Reply:
ok so turns out my assumption about the user id attached to the metadata is wrong. so i'm going to merge what i have for now since it works and isn't n^2, and we can talk about refactoring later. a good solution maybe would be to add both senderUserId and receiverUserId to the metadata and then we can use whichever one on the client we need.